### PR TITLE
chore: ownership change internal

### PR DESCRIPTION
**Pull request by AlexRiedler**
#### Why 

appsec shouldn't own this; I think it might be archived but should be removed.

### CODE REVIEW

**Comment by kathrynkodama** [comment]
@AlexRiedler we updated the codeowners to @wealthsimple/appsec-posture because platform-security used to own this gem and was listed as the previous codeowner. We have started to scope out decommissioning.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,2 +1,2 @@
-* @wealthsimple/appsec-posture
-.github/workflows/* @wealthsimple/developer-tools @wealthsimple/appsec-posture
+* @wealthsimple/developer-tools
+.github/workflows/* @wealthsimple/developer-tools
```
Comment (rperryng) [inline .github/CODEOWNERS:2]: this should be `@wealthsimple/backend-platform` otherwise PRs will ping FEPLAT as well

Comment (AlexRiedler) [inline .github/CODEOWNERS:2]:
```suggestion
* @wealthsimple/developer-tools
.github/workflows/* @wealthsimple/backend-platform
```

**Review by andrewthauer — APPROVED** [review]

**Comment by alisaifee** [comment]
@AlexRiedler can we close the PR?

**Comment by AlexRiedler** [comment]
sure, lets archive then!

**Pull request closed without merging**